plugins/strategy/action.go

```diff
@@ -70,7 +70,7 @@ func (a *Action) PushReason(r string) {
 		}
 	}
 
-	// Append current reason to history and update action
-	a.Meta[metaKeyReasonHistory] = append(history, a.Reason)
+	// Append current reason to history and update action.
```

Comment:
my only question would be whether we should bother duplicating `a.Reason` in the reason history?